core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -407,13 +408,23 @@ public CreateIndexRequest source(Map<String, ?> source) {
                     } catch (IOException e) {
                         throw new ElasticsearchParseException("failed to parse custom metadata for [{}]", name);
                     }
+                } else {
+                    // found a key which is neither custom defined nor one of the supported ones
+                    if (unsupportedKey == null) {
```

Comment:
Documentation was removed for this back in 1.0 but yes, some people may still be using it.  We should add deprecation logging now, and throw an exception in 6.0